### PR TITLE
Add events in send email action

### PR DESCRIPTION
**Pull request by c0nstantx**
Three new events are added in the send email process:

- RESETTING_SEND_EMAIL_INITIALIZE
- RESETTING_SEND_EMAIL_CONFIRM
- RESETTING_SEND_EMAIL_COMPLETED

in order to alter controller's response through listener(s)

### CODE REVIEW

File: Event/GetResponseNullableUserEvent.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+{
+    private $request;
+    private $user;
+
+    public function __construct(Request $request, UserInterface $user = null)
+    {
+        $this->user = $user;
+        $this->request = $request;
+    }
+}
```
Comment (OskarStark) [inline Event/GetResponseNullableUserEvent.php]: missing new line

File: Event/GetResponseNullableUserEvent.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * Description of GetResponseNullableUserEvent
+ *
+ * @author K.Christofilos <kostas.christofilos@rocketgraph.com>
+ */
+class GetResponseNullableUserEvent extends GetResponseUserEvent
+{
+    private $request;
```
Comment (OskarStark) [inline Event/GetResponseNullableUserEvent.php]: please add PHPDoc

File: Event/GetResponseNullableUserEvent.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * This file is part of Rocketgraph service
+ * <http://www.rocketgraph.com>
+ */
+
+namespace FOS\UserBundle\Event;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * Description of GetResponseNullableUserEvent
```
Comment (OskarStark) [inline Event/GetResponseNullableUserEvent.php]: useless generated comment, please remove

File: Event/GetResponseNullableUserEvent.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<?php
+/**
```
Comment (OskarStark) [inline Event/GetResponseNullableUserEvent.php]: wrong header, please use this one
```php
 /*
  * This file is part of the FOSUserBundle package.
  *
  * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
  *
  * For the full copyright and license information, please view the LICENSE
  * file that was distributed with this source code.
  */
`

File: Event/GetResponseNullableUserEvent.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ * Description of GetResponseNullableUserEvent
+ *
+ * @author K.Christofilos <kostas.christofilos@rocketgraph.com>
+ */
+class GetResponseNullableUserEvent extends GetResponseUserEvent
+{
+    private $request;
+    private $user;
+
+    public function __construct(Request $request, UserInterface $user = null)
```
Comment (OskarStark) [inline Event/GetResponseNullableUserEvent.php]: please add PHPDoc

File: Event/GetResponseNullableUserEvent.php [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * Description of GetResponseNullableUserEvent
+ *
+ * @author K.Christofilos <kostas.christofilos@rocketgraph.com>
+ */
+class GetResponseNullableUserEvent extends GetResponseUserEvent
+{
+    private $request;
+    private $user;
```
Comment (OskarStark) [inline Event/GetResponseNullableUserEvent.php]: please add PHPDoc

File: Controller/ResettingController.php [hunk 7]
```diff
@@ -46,6 +47,16 @@ public function sendEmailAction(Request $request)
 
         /** @var $user UserInterface */
         $user = $this->get('fos_user.user_manager')->findUserByUsernameOrEmail($username);
+        /** @var $dispatcher \Symfony\Component\EventDispatcher\EventDispatcherInterface */
```
Comment (OskarStark) [inline Controller/ResettingController.php]: please use a use statement and not an FQN

**Comment by OskarStark** [comment]
After my comments, this PR looks good and I like the new events 👍

**Comment by OskarStark** [comment]
LGTM 👍 

@XWB we should squash the commits to get rid of the merge commit

**Comment by XWB** [comment]
Thanks!

**Comment by c0nstantx** [comment]
I am really happy to contribute 😄

**Pull request merged**